zerver/tests/test_message_edit.py

```diff
@@ -2288,6 +2299,51 @@ def test_moving_all_topic_messages(
         # Test editing both topic and stream together.
         test_moving_all_topic_messages(new_topic="edited", new_stream=verona)
 
+        # Move these messages to the original stream and topic to test the next case.
+        self.client_patch(
+            f"/json/messages/{id4}",
+            {
+                "stream_id": denmark.id,
+                "topic": old_topic,
+                "propagate_mode": "change_all",
+                "send_notification_to_new_thread": "false",
+            },
+        )
+
+        # Test editing both topic and stream with no limit set.
+        do_set_realm_property(
+            user_profile.realm,
+            "move_messages_within_stream_limit_seconds",
+            None,
+            acting_user=None,
+        )
+        do_set_realm_property(
+            user_profile.realm,
+            "move_messages_between_streams_limit_seconds",
+            None,
+            acting_user=None,
+        )
+
+        new_stream = verona
+        new_topic = "edited"
+        result = self.client_patch(
+            f"/json/messages/{id4}",
+            {
+                "topic": new_topic,
+                "stream_id": new_stream.id,
+                "propagate_mode": "change_all",
+                "send_notification_to_new_thread": "false",
+            },
+        )
+        self.assert_json_success(result)
+        # Check message count in old topic and/or stream.
+        messages = get_topic_messages(user_profile, old_stream, old_topic)
+        self.assert_length(messages, 0)
+
+        # Check message count in new topic and/or stream.
+        messages = get_topic_messages(user_profile, new_stream, new_topic)
+        self.assert_length(messages, 5)
```

Comment:
I am not sure whether it is worth to add tests for a topic-only and stream-only edit for this case when there is no time limit set.

Reply:
Yeah it's probably unnecessary.